src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java

```diff
@@ -107,8 +108,15 @@ public CommandResult execute(Model model) throws CommandException {
         targetDoctor.addAppointment(toAdd);
 
         model.addAppointment(toAdd);
-        Timeslot timeslotToRemove = new Timeslot(toAdd.getDateTime().toLocalDate(), toAdd.getDateTime().getHour());
-        model.removeAvailableTimeSlot(timeslotToRemove);
+        if (model.getAvailableTimeSlotList().size() > 0) {
+            LocalDate apptDate = toAdd.getDateTime().toLocalDate();
+            LocalDate currDate = model.getAvailableTimeSlotList().get(0).getDate();
+            if (apptDate == currDate) {
```

Comment:
This will never be evaluate to true. Use equals(). Please test your code before opening PR next time.

Reply:
Fixed in latest commit. Great catch!